src/DynamoCoreWpf/ViewModels/Watch3D/DynamoPointLineMaterialVariable.cs

```diff
@@ -165,8 +166,115 @@ public override MaterialVariable CreateMaterialVariables(IEffectsManager manager
             return new DynamoPointLineMaterialVariable(manager, technique, MaterialVariable.DefaultPointLineConstantBufferDesc, this);
         }
     }
+    */
 
     internal class DynamoPointLineCore : PointLineRenderCore
     {
+
+        private bool isFrozenData;
+
+        /// <summary>
+        /// Is this model Frozen.
+        /// </summary>
+        public bool IsFrozenData { get { return isFrozenData; } internal set { SetAffectsRender(ref isFrozenData, value); } }
```

Comment:
Is there something special about doing `SetAffectsRender`? I've seen `Set` used in some places and another `Set`something method in helix.

Reply:
I see that you didn't have to override `OnUpdateModelStruct`; I remember you mentioned it doesn't get called anyway. Secondly, I'm still a little confused why you didn't need to add the geometry shader to the shader passes.

Reply:
`OnUpdateModelStruct` now gets called since the pipeline is setup correctly, unfortunately the point model struct is very small and doesn't have anywhere we can stick our extra data.

it's only 2 bools and a matrix that we need to do the projection - so I just used a value that is set to the PointLine constant buffer, I don't know how to update this otherwise, but it looks like it is somehow hooked into the default material inside the helix implementation ,but fortunately we can access it from the geometryModel object directly, so no need to hook into the render events etc.

I still really want to know how to add arbitrary data instead of reusing these unused variables, hopefully we can get more info from the helix maintainers...